src/metal.cs

```diff
@@ -2050,5 +2401,242 @@ interface MTLStageInputOutputDescriptor : NSCopying
+
+		[Export ("stopCapture")]
+		void StopCapture ();
+
+		[NullAllowed, Export ("defaultCaptureScope", ArgumentSemantic.Strong)]
+		IMTLCaptureScope DefaultCaptureScope { get; set; }
+
+		[Export ("isCapturing")]
+		bool IsCapturing { get; }
+	}
+
+	[Mac (10, 13, onlyOn64: true), NoiOS, NoTV, NoWatch]
+	enum MTLDeviceNotificationName {
+		[Field ("MTLDeviceWasAddedNotification")]
+		DeviceWasAdded,
+
+		[Field ("MTLDeviceRemovalRequestedNotification")]
+		DeviceRemovalRequested,
+
+		[Field ("MTLDeviceWasRemovedNotification")]
+		MTLDeviceWasRemoved,
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPipelineBufferDescriptor : NSCopying
+	{
+		[Export ("mutability", ArgumentSemantic.Assign)]
+		MTLMutability Mutability { get; set; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPipelineBufferDescriptorArray
+	{
+		[Export ("objectAtIndexedSubscript:")]
+		MTLPipelineBufferDescriptor GetObject (nuint bufferIndex);
+
+		[Export ("setObject:atIndexedSubscript:")]
+		void SetObject ([NullAllowed] MTLPipelineBufferDescriptor buffer, nuint bufferIndex);
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLArgumentDescriptor : NSCopying
+	{
+		[Static]
+		[Export ("argumentDescriptor")]
+		MTLArgumentDescriptor ArgumentDescriptor { get; }
+
+		[Export ("dataType", ArgumentSemantic.Assign)]
+		MTLDataType DataType { get; set; }
+
+		[Export ("index")]
+		nuint Index { get; set; }
+
+		[Export ("arrayLength")]
+		nuint ArrayLength { get; set; }
+
+		[Export ("access", ArgumentSemantic.Assign)]
+		MTLArgumentAccess Access { get; set; }
+
+		[Export ("textureType", ArgumentSemantic.Assign)]
+		MTLTextureType TextureType { get; set; }
+
+		[Export ("constantBlockAlignment")]
+		nuint ConstantBlockAlignment { get; set; }
+	}
+
+	interface IMTLArgumentEncoder { }
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[Protocol]
+	interface MTLArgumentEncoder
+	{
+		[Abstract]
+		[Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+
+		[Abstract]
+		[Export ("encodedLength")]
+		nuint EncodedLength { get; }
+
+		[Abstract]
+		[Export ("alignment")]
+		nuint Alignment { get; }
+
+		[Abstract]
+		[Export ("setArgumentBuffer:offset:")]
+		void SetArgumentBuffer (IMTLBuffer argumentBuffer, nuint offset);
+
+		[Abstract]
+		[Export ("setArgumentBuffer:startOffset:arrayElement:")]
+		void SetArgumentBuffer ([NullAllowed] IMTLBuffer argumentBuffer, nuint startOffset, nuint arrayElement);
+
+		[Abstract]
+		[Export ("setBuffer:offset:atIndex:")]
+		void SetBuffer ([NullAllowed] IMTLBuffer buffer, nuint offset, nuint index);
+
+		[Abstract]
+		[Export ("setBuffers:offsets:withRange:")]
+		void SetBuffers (IMTLBuffer[] buffers, IntPtr offsets, NSRange range);
```

Comment:
For this one, the selector parameter is ```const (NSUInteger *) offsets```

When I asked about it in slack, sebastien said:
```@risi there’s already similar cases in metal.cs

it’s using `IntPtr` to avoid allocations. If you use metal then you care for performance (and likely won’t mind using unsafe code)```

(and I do see several other bindings using IntPtr for offsets parameters).